model/export/AssemblyExporterService.php

```diff
@@ -108,29 +140,25 @@ protected function doExportCompiledDelivery($path, core_kernel_classes_Resource
             'label' => $compiledDelivery->getLabel(),
             'version' => $taoDeliveryVersion
         );
-        $directories = $compiledDelivery->getPropertyValues(new core_kernel_classes_Property(DeliveryAssemblyService::PROPERTY_DELIVERY_DIRECTORY));
+        $directories = $compiledDelivery->getPropertyValues($this->getProperty(DeliveryAssemblyService::PROPERTY_DELIVERY_DIRECTORY));
         foreach ($directories as $id) {
-            $directory = tao_models_classes_service_FileStorage::singleton()->getDirectoryById($id);
-            $files = $directory->getIterator();
-            foreach ($files as $file) {
-                $source = $this->getFileSource($directory, $file);
-                tao_helpers_File::addFilesToZip($zipArchive, $source, $directory->getRelativePath() . $file);
+            $directory = $this->getServiceLocator()->get(ServiceFileStorage::SERVICE_ID)->getDirectoryById($id);
+            foreach ($this->getAssemblyFilesReader()->getFiles($directory) as $filePath => $fileStream) {
+                tao_helpers_File::addFilesToZip($zipArchive, $fileStream, $filePath);
             }
-            $data['dir'][$id] = $directory->getRelativePath();
+            $data['dir'][$id] = $directory->getPrefix();
         }
 
-        $runtime = $compiledDelivery->getUniquePropertyValue(new core_kernel_classes_Property(DeliveryAssemblyService::PROPERTY_DELIVERY_RUNTIME));
+        $runtime = $compiledDelivery->getUniquePropertyValue($this->getProperty(DeliveryAssemblyService::PROPERTY_DELIVERY_RUNTIME));
         $serviceCall = tao_models_classes_service_ServiceCall::fromResource($runtime);
         $data['runtime'] = base64_encode($serviceCall->serializeToString());
 
-        $rdfExporter = new tao_models_classes_export_RdfExporter();
-        $rdfdata = $rdfExporter->getRdfString(array($compiledDelivery));
-        if (!$zipArchive->addFromString('delivery.rdf', $rdfdata)) {
+        $rdfData = $this->rdfExporter->getRdfString(array($compiledDelivery));
+        if (!$zipArchive->addFromString('delivery.rdf', $rdfData)) {

```

Comment:
probably 'delivery.rdf' should be a constant?

Reply:
Done.